### PR TITLE
Export queue rewrite and refactored code to use 'uri'

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: modules/core/src/main/java/webindex/core/IndexClient.java [hunk 1]
```diff
@@ -231,4 +249,79 @@ public Links getLinks(String rawUrl, String linkType, String next, int pageNum)
     }
     return links;
   }
+
+  public static Collection<Mutation> genDomainMutations(DomainUpdate update, long seq) {
+    Map<RowColumn, Bytes> oldData = genDomainData(update.getDomain(), update.getOldPageCount());
+    Map<RowColumn, Bytes> newData = genDomainData(update.getDomain(), update.getNewPageCount());
+    return AccumuloExportQueue.generateMutations(seq, oldData, newData);
```
Comment (keith-turner) [inline modules/core/src/main/java/webindex/core/IndexClient.java]: I did not realize this while looking at the recipes PR, but looking at this PR made me realize that generateMutations is a really generic name.

File: modules/core/src/main/java/webindex/core/models/UriInfo.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+package webindex.core.models;
+
+import java.io.Serializable;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Used by URI collision free map
+ */
+public class UriInfo implements Serializable {
```
Comment (keith-turner) [inline modules/core/src/main/java/webindex/core/models/UriInfo.java]: was this just copied out of UriMap? just wondering if there are any changes to review

File: modules/data/src/main/java/webindex/serialization/WebindexKryoFactory.java [hunk 3]
```diff
@@ -38,12 +39,12 @@ public Kryo create() {
     // same order it would be ok) and ran into issue where Spark and Fluo code were using different
     // ids for some reason.
     kryo.register(UriInfo.class, 9);
-    kryo.register(DomainExport.class, 10);
-    kryo.register(UriCountExport.class, 11);
-    kryo.register(PageExport.class, 12);
```
Comment (keith-turner) [inline modules/data/src/main/java/webindex/serialization/WebindexKryoFactory.java]: need to update docs/code-guide.md for these changes

File: modules/data/src/main/java/webindex/data/fluo/PageObserver.java [hunk 4]
```diff
@@ -24,27 +26,28 @@
...
 import org.apache.fluo.recipes.core.map.CollisionFreeMap;
 import org.apache.fluo.recipes.core.types.TypedSnapshotBase.Value;
 import org.apache.fluo.recipes.core.types.TypedTransactionBase;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import webindex.core.Constants;
 import webindex.core.models.Link;
 import webindex.core.models.Page;
+import webindex.core.models.UriInfo;
+import webindex.core.models.update.IndexUpdate;
```
Comment (keith-turner) [inline modules/data/src/main/java/webindex/data/fluo/PageObserver.java]: is everything in models.update pkg related to exporting?  If so I would prefer calling it models.export... but thats just my personal preference.  Update also make sense, I am just thinking export is more specific.

File: modules/data/src/main/java/webindex/data/fluo/IndexExporter.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  protected Collection<Mutation> processExport(SequencedExport<String, IndexUpdate> export) {
+    if (export.getValue() instanceof DomainUpdate) {
+      return IndexClient.genDomainMutations((DomainUpdate) export.getValue(), export.getSequence());
+    } else if (export.getValue() instanceof PageUpdate) {
+      return IndexClient.genPageMutations((PageUpdate) export.getValue(), export.getSequence());
+    } else if (export.getValue() instanceof UriUpdate) {
+      return IndexClient.genUriMutations((UriUpdate) export.getValue(), export.getSequence());
+    }
+
+    log.error("Unknown IndexUpdate object placed on index export queue");
```
Comment (mikewalch) [inline modules/data/src/main/java/webindex/data/fluo/IndexExporter.java]: Fixed in 5d58150b2be3a

File: modules/core/src/main/java/webindex/core/IndexClient.java (hunk 1 above)
Comment (mikewalch) [inline modules/core/src/main/java/webindex/core/IndexClient.java]: I agree.  Do you want to create an issue?

File: modules/data/src/main/java/webindex/data/fluo/PageObserver.java (hunk 4 above)
Comment (mikewalch) [inline modules/data/src/main/java/webindex/data/fluo/PageObserver.java]: I wanted to group them but I am not stuck on 'update'.  I can make the package 'export'

File: modules/core/src/main/java/webindex/core/IndexClient.java (hunk 1 above)
Comment (keith-turner) [inline modules/core/src/main/java/webindex/core/IndexClient.java]: opened apache/incubator-fluo-recipes#108

File: modules/core/src/main/java/webindex/core/models/UriInfo.java (hunk 2 above)
Comment (mikewalch) [inline modules/core/src/main/java/webindex/core/models/UriInfo.java]: yes it was just copied out

File: modules/data/src/main/java/webindex/serialization/WebindexKryoFactory.java (hunk 3 above)
Comment (mikewalch) [inline modules/data/src/main/java/webindex/serialization/WebindexKryoFactory.java]: updated code guide in f0cbafb014b1445

File: modules/data/src/main/java/webindex/data/fluo/PageObserver.java (hunk 4 above)
Comment (mikewalch) [inline modules/data/src/main/java/webindex/data/fluo/PageObserver.java]: change package name to 'export' in f0cbafb014b1445

**Comment by keith-turner** [comment]
+1

**Pull request merged**